lib/mail_retriever.rb

```diff
@@ -0,0 +1,73 @@
+require 'mail'
+
+# USAGE:
+#
+# manager = MailRetriever.new(
+#   config: {
+#     user_name: 'reporting@pensionwise.gov.uk',
+#     password: <password>
+#   }
+# )
+#
+# PASSWORD:
+#
+# Generate an app specific password following insteructions:
+#   https://support.google.com/mail/answer/185833?hl=en&rd=1
+#
+# manager.all(
+#   search_keys: [
+#     'SUBJECT', 'TP Daily Call Data',
+#     'FROM', '@pensionwise.gov.uk',
+#   ],
+#   file_name_regexp: /DWP Daily Data File.*\.xlsx/
+# )
+#
+
+class MailRetriever
+  DEFAULT_CONFIG = {
+    address: 'imap.googlemail.com',
+    port: 993,
+    enable_ssl: true,
+    user_name: nil,
+    password: nil
+  }.freeze
+
+  def initialize(config:)
+    Mail.defaults do
+      retriever_method :imap, DEFAULT_CONFIG.merge(config)
+    end
+  end
+
+  def most_recent(search_keys: ['All'], file_name_regexp:)
```

Comment:
This is done, may be worth adding a comment that search string needed to be quote enclosed if multiple words as otherwise you get a rather unhelpful error message.